src/gluonts/model/lstnet/_estimator.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+from typing import Optional
+
+# Third-party imports
+import numpy as np
+from mxnet.gluon import HybridBlock, loss
+
+# First-party imports
+from gluonts.core.component import validated, DType
+from gluonts.model.estimator import GluonEstimator
+from gluonts.model.lstnet._network import LSTNetTrain, LSTNetPredict
+from gluonts.model.predictor import Predictor, RepresentableBlockPredictor
+from gluonts.trainer import Trainer
+from gluonts.dataset.field_names import FieldName
+from gluonts.support.util import copy_parameters
+from gluonts.transform import (
+    Chain,
+    ExpectedNumInstanceSampler,
+    InstanceSplitter,
+    Transformation,
+    AsNumpyArray,
+    AddObservedValuesIndicator,
+)
+
+
+class LSTNetEstimator(GluonEstimator):
+    """
+    Constructs an LSTNet estimator for multivariate time-series data.
+
+    The model has been described in this paper:
+    https://arxiv.org/abs/1703.07015
+
+    Note that this implementation will change over time and we furthre work on
+    this method.
+
+    Parameters
+    ----------
+    num_series
+        Number of time-series (variats)
+    channels
+        Number of channels for first layer Conv1D
+    kernel_size
+        Kernel size for first layer Conv1D
+    skip_size
+        Skip size for the skip RNN layer
+    freq
+        Frequency of the data to train and predict
+    ar_window
+        Auto-regressive window size for the linear part
+    trainer
+        Trainer object to be used
+    context_length
+        The maximum number of steps to unroll the RNN for computing the predictions
+        (Note that it is constraints by the Conv1D output size)
+    prediction_length
+        Length of the prediction horizon
```

Comment:
If I remember correctly, in the paper the authors describe how the model produces forecasts for a single time step, at a predefined lead time. How does the model here produce the forecast for the entire horizon?

Reply:
It doesn't provide predictions for an entire horizon, only the very last time in the horizon Y_{t+h} by slicing.